Snowflake.Data/Core/FileTransfer/SFFileTransferAgent.cs

```diff
@@ -667,155 +667,150 @@ private void initEncryptionMaterial()
         }
 
         /// <summary>
-        /// Expand the expand the wildcards if any to generate the list of paths for all files 
-        /// matched by the wildcards. Also replace 
-        /// Get the absolute path for the file.
+        /// Expand the wildcards if any to generate the list of paths for all files matched by the wildcards.
+        /// Also replace the relative paths to the absolute paths for the files if needed.
         /// </summary>
         /// <param name="location">The path to expand</param>
         /// <returns>The list of file matching the input location</returns>
         /// <exception cref="DirectoryNotFoundException">Directory not found. Could not find a part of the pat </exception>
         /// <exception cref="FileNotFoundException">File not found or the path is pointing to a Directory</exception>
         private List<string> expandFileNames(string location)
         {
-            // Replace ~ with the user home directory path
-            if (location.Contains("~"))
-            {
-                string homePath = (Environment.OSVersion.Platform == PlatformID.Unix ||
-                Environment.OSVersion.Platform == PlatformID.MacOSX)
-                ? Environment.GetEnvironmentVariable("HOME")
-                : Environment.ExpandEnvironmentVariables("%HOMEDRIVE%%HOMEPATH%");
-
-                location = location.Replace("~", homePath);
-            }
-
-
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
-            {
-                location = Path.GetFullPath(location);
-            }
-
-            String fileName = Path.GetFileName(location);
-            string directoryName = Path.GetDirectoryName(location);
-
-            List<string> filePaths = new List<string>();
-            //filePaths.Add(""); //Start with an empty string to build upon
-            if (directoryName.Contains("?") ||
-                directoryName.Contains("*"))
-            {
-                // If there is a wildcard in at least one of the directory name in the file path
-                string[] pathParts = location.Split(Path.DirectorySeparatorChar);
-
-                string currPart;
-                for (int i = 0; i < pathParts.Length; i++)
+            location = ExpandHomeDirectoryIfNeeded(location);
+            var fileName = Path.GetFileName(location);
+            var directoryName = Path.GetDirectoryName(location);
+            var foundDirectories = ExpandDirectories(directoryName);
+            var filePaths = new List<string>();
+            
+            if (ContainsWildcard(fileName))
+            {
+                foreach (var directory in foundDirectories)
                 {
-                    List<string> tempPaths = new List<string>();
-                    foreach (string filePath in filePaths)
+                    var ext = Path.GetExtension(fileName);
+                    if (4 == ext.Length && fileName.Contains('*'))
```

Comment:
Could you please comment on value 4 here?

Reply:
Added a comment here